### ISSUE TITLE
align strings when trailing backslash is used

### ISSUE THREAD

**Issue by nikomatsakis** [rust-lang-nursery/rustfmt#1026]
Rust has this nifty feature that a trailing backslash ignores leading whitespace on the next line when found in a string constant. This means that if I have a Rust source file like:

``` rust
fn main() {
    println!("foo\
bar");
}
```

I can instead indent it like:

``` rust
fn main() {
    println!("foo\
              bar");
}
```

But rustfmt doesn't do this, from what I can tell. Perhaps we ought to?

**Comment by marcusklaas** [comment]
We used to do this by default, but we have disabled it because it often did not do _the right thing_.. I'm sure there's a configuration option for this.

**Comment by nikomatsakis** [comment]
@marcusklaas hmm, are there older issues yo can point me at? in what way could it not do the right thing? it seems like if people don't want the strings aligned, they should not use `\`...

**Comment by topecongiro** [comment]
Closed via #1724.
